keras/losses.py

```diff
@@ -39,6 +39,12 @@ def hinge(y_true, y_pred):
     return K.mean(K.maximum(1. - y_true * y_pred, 0.), axis=-1)
 
 
+def huber(y_true, y_pred, huber_delta=0.1):
+    x = K.abs(y_true - y_pred)
+    x = K.switch(x < huber_delta, 0.5 * x**2, huber_delta * (x - 0.5 * huber_delta))
```

Comment:
Could you use a different variable here instead of x. Don't think relational operator is supported between instances of 'Function' and 'float' value.

Reply:
I have changed the code, and now, code is compatible to all backend types and also resolved issues related to it.
Please merge it if you find it useful.